waiter/src/waiter/async_request.clj

```diff
@@ -153,7 +153,7 @@
     ;; trigger execution of monitoring system
     (letfn [(make-get-request-fn []
               (counters/inc! (metrics/service-counter service-id "request-counts" "async-monitor"))
-              (let [request-stub {:body nil, :headers {}, :request-method :get}]
+              (let [request-stub {:body nil :headers {} :query-string query-string :request-method :get}]
```

Comment:
Why is the `query-string` being passed?  Are we supposed to be checking what's in the `Location` header (which can include its own query-string) from the response?  Why should the `query-string` from the original request be involved into this process?

Reply:
We extract the query-string from the location header now. Added unit test to verify this behavior.